### DISCUSSION TITLE
Where does Chia stored wallet private keys?

### DISCUSSION THREAD

**Discussion by keeperlink** [Chia-Network/chia-blockchain#6281] [Q&A]
Where does Chia store wallet private keys?
Is it in .chia/wallet/ directory?

**Comment by kettlewarrior** [comment]
Looks like here:

```
cd ~/.chia/mainnet/config/ssl/wallet
```

**Reply by keeperlink** [reply]
right. looks like it. thank you

**Reply by room101-dev** [reply]
You know the simple cmd "chia keys show --mneumonic" allows anybody on RPC to dump your private, but if you delete this file (back it up), then you priv-keys aren't available.

Also for mining you don't need the wallet, you just really need it for 'address generation', so its probably best practice to have an offline 'chia start wallet' process running with this file, and make sure that its on an offline system, and keep lots of backup copys of your wallet, or just run  "chia keys show --mneumonic", and print. IMHO the reason they left this secret stupid command in the code is because the contract programmers of chia in INDIA, want to rip off western morons forever, there is no other explanation for this security whole

CHIA was designed to enable criminals.
